code/modules/fishing/fishing_gear.dm

```diff
@@ -350,3 +350,341 @@ TYPEINFO(/obj/item/fish_portal)
+	hit_type = DAMAGE_STAB
+	flags = FPRINT | TABLEPASS | USEDELAY
+	w_class = W_CLASS_NORMAL
+	force = 10
+	throwforce = 5
+	throw_speed = 1
+	throw_range = 5
+	contraband = 4
+	is_syndicate = TRUE
+	var/obj/item/implant/syndie_lure/lure = null
+	// delay between tossing or reeling or etc
+	var/usage_cooldown = 0.8 SECONDS
+	// time per step to reel/filet a mob
+	var/syndie_fishing_speed = 0.7 SECONDS
+	// cooldown after throwing a hooked target around
+	var/yank_cooldown = 6 SECONDS
+	// how far you throw when yanking them
+	var/yank_range = 4
+	// how far the line can stretch
+	var/line_length = 9
+	// true if the rod is currently ""fishing"", false if it isnt
+	var/is_fishing = FALSE
+	HELP_MESSAGE_OVERRIDE({"The Glaucus starts with 7 damage on a melee reel, but builds up 3 additional damage on each ranged reel. If this reaches <b>25 damage</b>, or 6 ranged reels before a melee reel, the target will be stunned when damaged."})
+
+	New()
+		..()
+		src.lure = new (src)
+		src.lure.rod = src
+
+	get_desc(dist)
+		..()
+		if (dist < 1) // on our tile or our person
+			if (.) // we're returning something
+				. += " " // add a space
+			if (!src.lure)
+				src.lure = new (src)
+				src.lure.rod = src
+			. += "There is \a [src.lure.name] presented as bait."
+
+	attackby(obj/item/I, mob/user)
+		if (!src.lure)
+			src.lure = new (src)
+			src.lure.rod = src
+		if (src.lure.loc == src)
+			boutput(user, "You scan \the [I.name] onto \the [src.name]'s holographic bait projector.")
+			src.lure.name = I.name
+			src.lure.real_name = I.name
+			src.lure.desc = I.desc
+			src.lure.real_desc = I.desc
+			src.lure.appearance = I
+			src.lure.set_dir(I.dir)
+			src.lure.overlay_refs = I.overlay_refs?.Copy()
+			src.lure.plane = initial(src.lure.plane)
+			src.lure.layer = initial(src.lure.layer)
+			src.lure.tooltip_rebuild = 1
+			tooltip_rebuild = 1
+		else
+			boutput(user, "You can't change the bait while the line is out!")
+		return
+
+	attack_self(mob/user)
+		. = ..()
+		if (!src.lure)
+			src.lure = new (src)
+			src.lure.rod = src
+		if (src.lure.loc == src)
+			boutput(user, "You clean \the [src.name]'s holographic bait projector.")
+			src.lure.clean_forensic()
+		else
+			if(!src.lure.owner)
+				if(BOUNDS_DIST(src.lure,src) == 0)
+					src.is_fishing = FALSE
+					src.UpdateIcon()
+					user.update_inhands()
+					src.lure.set_loc(src)
+				else
+					step_towards(src.lure, src)
+			else
+				user.visible_message("<span class='alert'><b>[user] yanks the lure out of [src.lure.owner]!</b></span>")
+				src.lure.set_loc(get_turf(src.lure.loc))
+				src.lure.on_remove(src.lure.owner)
+
+	// here so that afterattack is called at range, which is the least bad way to throw the lure i think maybe
+	pixelaction(atom/target, params, mob/user, reach)
+		..()
+
+	afterattack(atom/target, mob/user)
+		..()
+		if (!isturf(user.loc))
+			return
+		if (!src.lure)
+			src.lure = new (src)
+			src.lure.rod = src
+			tooltip_rebuild = 1
```

Comment:
other versions of this snippet don't set `tooltip_rebuild`, maybe consider making this a proc so it isn't inconsistent?